### ISSUE TITLE
Dimensional_oM: IElementC - Consumption based evaluations

### ISSUE THREAD

**Issue by michaelhoehn** [BHoM/BHoM#1194]
#### Description:

Capturing an issue arising from the recent MEP_oM work and the desire to be able to assess both the material properties of the elements, but also the consumptive properties as well. 

A light fixture is a simple example, I want to assess the material properties of the light fixture, but also understand the Wattage consumed over the course of a defined amount of time.

A more complex, but very important example would be something like a refrigerant-based MEP system, it has a material property for pipes being used, but hosts and consumes an amount of refrigerant over the course of years. 

There's a need for discussion on this topic prior to creation. It would be ideal to hear from the end-users to understand more clearly how they intend to use this functionality, and when it should be developed. I imagine there are lots of other applications across disciplines.  

Obviously, this is also touching on the concept of Time or the mysterious IElementT (Temporal). Although there isn't necessarily urgency behind this development, it would be good to get ahead of the concepts to define them together. 

TL;DR ElementC to facilitate consumption-based object properties.

**Comment by travispotterBH** [comment]
@michaelhoehn  Can you review this? Is this planned? It looks like MEP related, but no progression since 2021.
